src/Libraries/RevitNodes/Elements/Parameter.cs

```diff
@@ -25,12 +30,172 @@ public string Name
             }
         }
 
+        /// <summary>
+        /// Get the value of the parameter
+        /// </summary>
+        public object Value
+        {
+            get
+            {
+                switch (InternalParameter.StorageType)
+                {
+                    case Autodesk.Revit.DB.StorageType.Double: return InternalParameter.AsDouble();
+                    case Autodesk.Revit.DB.StorageType.Integer: return InternalParameter.AsInteger();
+                    case Autodesk.Revit.DB.StorageType.ElementId: return InternalParameter.AsElementId();
+                    case Autodesk.Revit.DB.StorageType.None: return null;
+                    default: return InternalParameter.AsString();
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Check if the Parameter has a value
+        /// </summary>
+        public bool HasValue
+        {
+            get { return InternalParameter.HasValue; }
+        }
+
+        /// <summary>
+        /// Check if the Parameter is read only
+        /// </summary>
+        public bool IsReadOnly
+        {
+            get { return InternalParameter.IsReadOnly; }
+        }
+
+        /// <summary>
+        /// Check if the Parameter is shared
+        /// </summary>
+        public bool IsShared
+        {
+            get { return InternalParameter.IsShared; }
+        }
+
+        /// <summary>
+        /// Get the parameter's group
+        /// </summary>
+        public string Group
+        {
+            get { return InternalParameter.Definition.ParameterGroup.ToString(); }
+        }
+
+        /// <summary>
+        /// Get the parameter type
+        /// </summary>
+        public string ParameterType
+        {
+            get { return InternalParameter.Definition.ParameterType.ToString(); }
+        }
+
+        /// <summary>
+        /// Get the parameter's element Id
+        /// </summary>
+        public ElementId GetParameterId
```

Comment:
ParameterId. Usually for public propertis, "Get" is not added in the front. Also shall we keep it consistent with the existing Element.Id?